### DISCUSSION TITLE
Automatic back-up to GoogleDrive or OneDrive with selectable frequency

### DISCUSSION THREAD

**Discussion by ZoePuffet** [iSoron/uhabits#1663] [Feature Requests]
My phone got stolen very recently, so I went back to my old phone. Then I noticed there is no way to get all the data/progress/added habits back unless I myself had gone into the settings and backed everything up manually.
Is there a feature I'm missing or are there plans to add something like this?

**Comment by cameronj86** [comment]
Can someone point me to the folder in question? I'm thinking of using [Autosync](https://play.google.com/store/apps/details?id=com.ttxapps.autosync) to take care of the backup process, but not sure of file locations on the back-end.

Found the automatic backup @ (/storage/Android/data/org.isoron.uhabits/files/Backups/) thanks to https://github.com/iSoron/uhabits/discussions/178#discussioncomment-5010107

Now I just need to figure out what folder to overwrite locally and point AutoSync to do it

**Comment by nikhilweee** [comment]
I would really really like this feature to be implemented. Habits are formed over weeks and months, and just because of an accidental uninstall I wouldn't want to lose all my progress. There has been discussion about automatic backups in #178 but the solution to that discussion only exports a backup to some local folder that would be deleted by an accidental uninstall. Automated cloud backups would be a life saver. @iSoron what do you think about this?
